tests/transparent.rs

```diff
@@ -0,0 +1,97 @@
+use snafu::prelude::*;
+use std::error::Error;
+
+#[derive(Debug, Snafu)]
+enum AlphaError {
+    AlphaDummy,
+}
+
+#[derive(Debug, Snafu)]
+enum BetaError {
+    BetaDummy { source: AlphaError },
+}
+
+#[derive(Debug, Snafu)]
+enum ErrorTest {
+    #[snafu(transparent)]
+    Alpha { source: AlphaError },
+
+    #[snafu(transparent)]
+    Beta { source: BetaError },
+}
+
+fn alpha() -> Result<i32, AlphaError> {
+    AlphaDummySnafu.fail()
+}
+
+fn beta() -> Result<i32, BetaError> {
+    let a = alpha().context(BetaDummySnafu)?;
+    Ok(a + 2)
+}
+
+fn example() -> Result<i32, ErrorTest> {
+    let a = 2;
+    let b = beta()?;
+    Ok(a * 10 + b)
+}
+
+fn check<T: std::error::Error>() {}
+
+#[test]
+fn implements_error() {
+    check::<ErrorTest>();
+
+    let error = example().unwrap_err();
+    // skpping `ErrorTest`, directly going yo BetaError's source
+    assert_eq!(error.source().unwrap().to_string(), "AlphaDummy");
+}
+
+#[test]
+fn implements_display() {
+    let error = example().unwrap_err();
+    // no `Beta: ` prefix
+    assert_eq!(error.to_string(), "BetaDummy: AlphaDummy");
+}
+
+mod with_backtraces {
+    use super::*;
+    use snafu::Backtrace;
+
+    #[derive(Debug, Snafu)]
+    enum Error {
```

Comment:
Changing this one to a struct will allow us to double-dip and have a test that `transparent` works on structs.